mcs/class/corlib/System.Globalization/CultureInfo.cs

```diff
@@ -146,17 +146,26 @@ struct Data {
 			}
 		}
 
+		private static string current_culture_name;
+
 		internal static CultureInfo ConstructCurrentCulture ()
 		{
 			if (default_current_culture != null)
 				return default_current_culture;
 
-			var locale_name = get_current_locale_name ();
+			string locale_name;
+			if (current_culture_name != null)
+				locale_name = current_culture_name;
+			else
+				locale_name = get_current_locale_name ();
+
 			CultureInfo ci = null;
 
 			if (locale_name != null) {
 				try {
 					ci = CreateSpecificCulture (locale_name);
```

Comment:
we're always reconstructing `ci` even though current_culture_name is set?

Reply:
Storing a single CultureInfo left me with the choice of making it readonly or not, 
https://msdn.microsoft.com/en-us/library/system.globalization.cultureinfo.readonly(v=vs.110).aspx

If I didn't, then I was concerned that there was some way to abuse this singleton we return. If I did make it readonly, that might break existing code that depends on it not being readonly. 

I picked a conservative choice. If you have assurances that it's safe to pick one of the above policies, I'd love to make a singleton.

Reply:
I do not know why we clear the cached objects while leaving the cached strings in unmanaged. I'm not sure what that accomplishes.

Reply:
I'm confused about two things:

1. Why is `default_current_culture` getting cleared? I can't find any callers of `ClearCachedData()`.
2. Why is `ConstructCurrentCulture()` even being called more than once. The only caller is `nativeInitCultureAccessors` and that in turn shouldn't be called more than once per thread.